### PR TITLE
Carthage support

### PR DESCRIPTION
**Pull request by neonichu**
This adds support for using Moya with Carthage, using the technique described in CocoaPods/pod-template#106. It merely makes it possible to build with Carthage, though, to be useful without requiring the user to have CP installed as well, the `Pods/` directory would need to be added to the repo. I didn't want to pollute the PR with that, though.

It is WIP, because it currently needs `master` of both CocoaPods and Carthage. I'll update the PR once those things are released, for now it is just for the adventurous.

The only duplicated effort this requires is declaring the dependencies in the `Cartfile` as well as the `Podfile`.

### CODE REVIEW

**Comment by ashfurrow** [comment]
Cool, thanks Boris!

**Comment by neonichu** [comment]
@ashfurrow since both CP as well as Carthage have shipped now with the required changes, this should be good to merge. 

Keep this in mind, though:

> the Pods/ directory would need to be added to the repo

**Comment by ashfurrow** [comment]
OK cool. I've got a PR open for checking in the Pods directory. Once that is :green_apple: , I'll merge and you can rebase from master? Or should it be safe to merge directly in?

/cc @MichaelMcGuire

File: Makefile [hunk 1]
```diff
@@ -13,9 +13,10 @@ clean:
 
 test:
 	cd Demo ; set -o pipefail && xcodebuild -workspace '$(WORKSPACE)' -scheme '$(SCHEME)' -configuration '$(CONFIGURATION)' test -sdk iphonesimulator -destination 'name=iPhone 5' | xcpretty -c --test
+	carthage build --no-skip-current
 
 setup:
-	cd Demo ; bundle install ; bundle exec pod install
+	cd Demo ; bundle install ; bundle exec pod install ; brew install carthage
```
Comment (ashfurrow) [inline Makefile]: Hmm. Setup is really for internal development, which someone is totally able to do with or without Carthage. Dunno that it needs to be here though.

**Comment by ashfurrow** [comment]
Ha, meeerrrrge conflict.

File: Makefile (hunk 1 above)
Comment (neonichu) [inline Makefile]: The goal was to do the `carthage build` as part of CI to catch issues with it

Comment (ashfurrow) [inline Makefile]: Ah, I see that. Makes sense.

**Comment by ashfurrow** [comment]
@neonichu Wanna rebase so I can merge?

**Comment by neonichu** [comment]
@ashfurrow should be good to merge now, sir!

**Comment by ashfurrow** [comment]
:red: can you `pod install` or `update` or whatever needs to pass?

**Comment by neonichu** [comment]
@ashfurrow this should be :ok_hand: now

**Pull request merged**